rholang/src/test/scala/coop/rchain/rholang/interpreter/SynchronousOutputSpec.scala

```diff
@@ -0,0 +1,52 @@
+package coop.rchain.rholang.interpreter
+
+import coop.rchain.rholang.interpreter.ParBuilderUtil.assertCompiledEqual
+import org.scalatest.{FlatSpec, Matchers}
+
+class SynchronousOutputSpec extends FlatSpec with Matchers {
+
+  "';'" should "have a higher precedence than '|'" in {
+    val s = "@1!?(1) ; @2!?(2). | @3!?(3)."
```

Comment:
I think it would be better to have another example like `@3!?(3). | @1!?(1) ; @2!?(2). ` which is a more convincing test case that ";" have higher precedence than "|" based on the perspective of seeing this program in sequence which ";" should consider first.
Does that make sense?